simulator/src/main/java/simblock/node/Farmer.java

```diff
@@ -0,0 +1,105 @@
+package simblock.node;
+
+import simblock.block.Block;
+import simblock.block.PoSpaceBlock;
+import simblock.simulator.Simulator;
+import simblock.simulator.Timer;
+import simblock.task.AbstractMintingTask;
+
+import java.math.BigInteger;
+
+import static simblock.settings.SimulationConfiguration.chia_k;
+import static simblock.simulator.Simulator.arriveBlock;
+
+public class Farmer extends Node{
+
+    private int[] PoSpaceCount;
+    private PoSpaceBlock[] Chain;
+    public Farmer(
+            int nodeID, int numConnection, int region, long miningPower, String routingTableName, boolean useCBR, boolean isChurnNode, int depth
+    ) {
+        super(nodeID, numConnection, region, miningPower, routingTableName, useCBR, isChurnNode);
+        PoSpaceCount = new int[depth];
+        Chain = new PoSpaceBlock[chia_k];
```

Comment:
The chia_k parameter specifies how many blocks an honest farmer should extend at every level. The Chain parameter keeps track of the three 'branches' that the farmer is extending. Each index holding a reference to the head of the branch which has a reference to it's parent block and so on.

Reply:
Based on discussion on Tuesday and email correspondence, I will replace Chain with a priority queue of some specified size. The queue will be used to keep track of the x best paths a Farmer is aware of